src/System.Security.Cryptography.ProtectedData/src/System/Security/Cryptography/ProtectedData.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+using Internal.Cryptography;
+
+using DATA_BLOB = Interop.Crypt32.DATA_BLOB;
+using CryptProtectDataFlags = Interop.Crypt32.CryptProtectDataFlags;
+
+namespace System.Security.Cryptography
+{
+    public static partial class ProtectedData
+    {
+        public static byte[] Protect(byte[] userData, byte[] optionalEntropy, DataProtectionScope scope)
+        {
+            if (userData == null)
+                throw new ArgumentNullException("userData");
+
+            return ProtectOrUnprotect(userData, optionalEntropy, scope, protect: true);
+        }
+
+        public static byte[] Unprotect(byte[] encryptedData, byte[] optionalEntropy, DataProtectionScope scope)
+        {
+            if (encryptedData == null)
+                throw new ArgumentNullException("encryptedData");
+
+            return ProtectOrUnprotect(encryptedData, optionalEntropy, scope, protect: false);
+        }
+
+        private static byte[] ProtectOrUnprotect(byte[] inputData, byte[] optionalEntropy, DataProtectionScope scope, bool protect)
+        {
+            Debug.Assert(inputData != null);
+
+            unsafe
```

Comment:
Since the whole body is unsafe, you could save a level of indentation and just add this to the signature.